Neuro/Tasks/TasksHandler.cs

```diff
@@ -174,28 +174,31 @@ private void Update()
         if (Minigame.Instance) return;
         if (!PlayerControl.LocalPlayer) return;
 
-        foreach (PlayerTask task in PlayerControl.LocalPlayer.myTasks)
+        //Open anything nearby, speedrun.
+        foreach (var tasky in PlayerControl.LocalPlayer.myTasks)
         {
-            if (task.IsComplete || task.Locations == null) continue;
+            if (tasky.IsComplete || tasky.Locations == null) continue;
 
-            // TODO: HOLY FUCK DON'T UNCOMMENT THIS THIS IS BROKEN AF
-            // INVOKE CONSOLE.USE DIRECTLY PLEEEEEEASE
+            Il2CppSystem.Collections.Generic.List<Console> list = tasky.FindConsoles();
 
-            /*// TODO: Invoke Console.Use directly and check if we can open it before trying
-            foreach (Vector2 location in task.Locations)
-            {
-                if (Vector2.Distance(location, PlayerControl.LocalPlayer.transform.position) < 0.8f)
+            foreach (Console con in list) {
+                var currentDistance = Vector2.Distance(con.transform.position, PlayerControl.LocalPlayer.transform.position);
+
+                if (currentDistance < con.usableDistance && !con.name.Contains("Vent"))
```

Comment:
Does this work with the vent cleaning minigame? I assume the exclude here is to prevent auto venting as an imposter but its unclear.

Reply:
> Does this work with the vent cleaning minigame? I assume the exclude here is to prevent auto venting as an imposter but its unclear.

It was leftover from async pathing testing, removed now.